src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1436,3 +1459,157 @@ def write_sorting(
+        Whether or not to overwrite the NWBFile if one exists at the nwbfile_path.
+        The default is False (append mode).
+    verbose: bool, optional
+        If 'nwbfile_path' is specified, informs user after a successful write operation.
+        The default is True.
+    unit_ids: list, optional
+        Controls the unit_ids that will be written to the nwb file. If None (default), all
+        units are written.
+    write_electrical_series: bool, optional
+        If True, the recording object associated to the WaveformExtractor is written as an electrical series.
+    write_electrical_series_kwargs: dict, optional
+        Keyword arguments to control the `write_recording()` function in case write_electrical_series=True
+    property_descriptions: dict
+        For each key in this dictionary which matches the name of a unit
+        property in sorting, adds the value as a description to that
+        custom unit column.
+    skip_properties: list of str
+        Each string in this list that matches a unit property will not be written to the NWBFile.
+    skip_features: list of str
+        Each string in this list that matches a spike feature will not be written to the NWBFile.
+    write_as: str (optional, defaults to 'units')
+        How to save the units table in the nwb file. Options:
+        - 'units' will save it to the official NWBFile.Units position; recommended only for the final form of the data.
+        - 'processing' will save it to the processing module to serve as a historical provenance for the official table.
+    units_name : str (optional, defaults to 'units')
+        The name of the units table. If write_as=='units', then units_name must also be 'units'.
+    units_description : str (optional)
+    """
+    if metadata is None:
+        metadata = dict()
+    assert (
+        nwbfile_path is None or nwbfile is None
+    ), "Either pass a nwbfile_path location, or nwbfile object, but not both!"
+    if nwbfile is not None:
+        assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be a pynwb.NWBFile object!"
+
+    # retrieve templates and stds
+    template_means = waveform_extractor.get_all_templates()
+    template_stds = waveform_extractor.get_all_templates(mode="std")
+    sorting = waveform_extractor.sorting
+    if unit_ids is not None:
+        unit_indices = sorting.ids_to_indices(unit_ids)
+        template_means = template_means[unit_indices]
+        template_stds = template_stds[unit_indices]
+
+    # metrics properties (quality, template) are temporarily added as properties
+    # to be written in the Units table
+    tmp_properties = []
+    if waveform_extractor.is_extension("quality_metrics"):
+        qm = waveform_extractor.load_extension("quality_metrics").get_data()
+        for prop in qm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, qm[prop])
+                tmp_properties.append(prop)
```

Comment:
Lines not covered by tests: https://app.codecov.io/gh/catalystneuro/neuroconv/blob/write-waveforms/src/neuroconv/tools/spikeinterface/spikeinterface.py